staticaddr/manager.go

```diff
@@ -0,0 +1,167 @@
+package staticaddr
+
+import (
+	"context"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/loop"
+	"github.com/lightninglabs/loop/staticaddr/script"
+	"github.com/lightninglabs/loop/swap"
+	staticaddressrpc "github.com/lightninglabs/loop/swapserverrpc"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+// ManagerConfig holds the configuration for the address manager.
+type ManagerConfig struct {
+	// AddressClient is the client that communicates with the loop server
+	// to manage static addresses.
+	AddressClient staticaddressrpc.StaticAddressServerClient
+
+	// LoopClient provides loop rpc functionality.
+	SwapClient *loop.Client
+
+	// Store is the database store that is used to store static address
+	// related records.
+	Store AddressStore
+
+	// Lnd is the lnd client that is used to derive new keys from lnd's
+	// wallet.
+	Lnd lndclient.LndServices
```

Comment:
Would be enough to just store the walletkit until we need more interfaces.

Reply:
We also need the chain params in this PR, and in the follow-up PRs the Chainnotifier and others.

What's the drawback in importing all of the `LndServices`? I like it cause it keeps the manager config more compact.